src/libifconfig.c

```diff
@@ -50,6 +54,31 @@
 
 #define NOTAG    ((u_short) -1)
 
+static bool
+isnd6defif(ifconfig_handle_t *h, const char *name)
+{
+	struct in6_ndifreq ndifreq;
+	unsigned int ifindex;
+	int s, error;
+
+	if (ifconfig_socket(h, AF_INET6, &s) != 0) {
+		return (-1);
+	}
+
+	memset(&ndifreq, 0, sizeof(ndifreq));
+	strlcpy(ndifreq.ifname, name, sizeof(ndifreq.ifname));
+	ifindex = if_nametoindex(ndifreq.ifname);
+	error = ioctl(s, SIOCGDEFIFACE_IN6, (caddr_t)&ndifreq);
+	if (error) {
```

Comment:
I think we should look at writing a function and/or macro to implement this pattern, because it's coming up a lot.